hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -190,15 +191,22 @@ public Record getRecord(Data key) {
     }
 
     @Override
-    public Record putReplicatedRecord(Data dataKey, Record replicatedRecord,
-                                      ExpiryMetadata expiryMetadata,
-                                      boolean populateIndexes, long now) {
-        Record newRecord = createRecord(replicatedRecord, now);
-        storage.put(dataKey, newRecord);
-        expirySystem.add(dataKey, expiryMetadata.getTtl(),
-                expiryMetadata.getMaxIdle(), expiryMetadata.getExpirationTime(),
-                now, expiryMetadata.getLastUpdateTime());
-        mutationObserver.onReplicationPutRecord(dataKey, newRecord, populateIndexes);
+    public Record putOrUpdateReplicatedRecord(Data dataKey, Record replicatedRecord,
+                                              ExpiryMetadata expiryMetadata,
+                                              boolean indexesMustBePopulated, long now) {
+        Record newRecord = storage.get(dataKey);
+        if (newRecord == null) {
+            newRecord = createRecord(replicatedRecord != null
+                    ? replicatedRecord.getValue() : null, now);
+            storage.put(dataKey, newRecord);
+        } else {
+            storage.updateRecordValue(dataKey, newRecord, replicatedRecord.getValue());
+            updateStatsOnPut(false, now);
```

Comment:
Minor: this stat update seems redundant to me since the last update time stat of record is already updated in `updateStatsOnPut(long hits, long now)` below, L:210.

Reply:
fixed